src/code/code_800430A0.c

```diff
@@ -1,24 +1,63 @@
 #include <ultra64.h>
 #include <global.h>
+#include <vt.h>
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_800430A0/func_800430A0.s")
+void func_800430A0(CollisionContext* colCtx, s32 bgId, Actor* actor) {
+    MtxF prevSrp;
+    MtxF prevSrpInv;
+    MtxF curSrp;
+    Vec3f pos;
+    Vec3f tempPos;
 
-void func_800432A0(CollisionContext* colCtx, u32 floorPolySource, Actor* actor) {
-    if (func_8003E934(floorPolySource) != 0) {
-        s16 v1 = colCtx->dyna.actorMeshArr[floorPolySource].rot2.y - colCtx->dyna.actorMeshArr[floorPolySource].rot1.y;
+    if (func_8003E934(bgId)) {
+        SkinMatrix_SetScaleRotateYRPTranslate(
+            &prevSrp, colCtx->dyna.bgActors[bgId].prevSrp.scale.x, colCtx->dyna.bgActors[bgId].prevSrp.scale.y,
+            colCtx->dyna.bgActors[bgId].prevSrp.scale.z, colCtx->dyna.bgActors[bgId].prevSrp.rot.x,
+            colCtx->dyna.bgActors[bgId].prevSrp.rot.y, colCtx->dyna.bgActors[bgId].prevSrp.rot.z,
+            colCtx->dyna.bgActors[bgId].prevSrp.pos.x, colCtx->dyna.bgActors[bgId].prevSrp.pos.y,
+            colCtx->dyna.bgActors[bgId].prevSrp.pos.z);
+        if (SkinMatrix_Invert(&prevSrp, &prevSrpInv) != 2) {
+            SkinMatrix_SetScaleRotateYRPTranslate(
+                &curSrp, colCtx->dyna.bgActors[bgId].curSrp.scale.x, colCtx->dyna.bgActors[bgId].curSrp.scale.y,
+                colCtx->dyna.bgActors[bgId].curSrp.scale.z, colCtx->dyna.bgActors[bgId].curSrp.rot.x,
+                colCtx->dyna.bgActors[bgId].curSrp.rot.y, colCtx->dyna.bgActors[bgId].curSrp.rot.z,
+                colCtx->dyna.bgActors[bgId].curSrp.pos.x, colCtx->dyna.bgActors[bgId].curSrp.pos.y,
+                colCtx->dyna.bgActors[bgId].curSrp.pos.z);
+            SkinMatrix_Vec3fMtxFMultXYZ(&prevSrpInv, &actor->posRot.pos, &tempPos);
+            SkinMatrix_Vec3fMtxFMultXYZ(&curSrp, &tempPos, &pos);
+            actor->posRot.pos = pos;
+            if (32760.0f <= pos.x || pos.x <= -32760.0f || 32760.0f <= pos.y || pos.y <= -32760.0f ||
+                32760.0f <= pos.z || pos.z <= -32760.0f) {
+
+                osSyncPrintf(VT_FGCOL(RED));
+                // @bug file and line are not passed to osSyncPrintf
+                osSyncPrintf(
+                    "BGCheckCollection_typicalActorPos():位置が妥当ではありません。\npos (%f,%f,%f) file:%s line:%d\n",
+                    pos.x, pos.y, pos.z);
+                // EUC-JP: 位置が妥当ではありません。 | Position is not valid
+                osSyncPrintf(VT_RST);
```

Comment:
you can take out the EUC-JP stuff and just put the english in a comment above the print